MLIR.cmake

```diff
@@ -19,27 +19,35 @@ include(TableGen)
 include(AddLLVM)
 include(AddMLIR)
 
+# When LLVM_ENABLE_ASSERTIONS is set to ON, the build is not Debug and the compiler
+# is clang, onnx_proto will fail to link to libprotobuf due to missing symbols when
+# we let HandleLLVMOptions actually set all the expected properties correctly (such
+# as removing NDEBUG from the compiler options). This appears to be a bug in the way
+# the libs interact with clang, because everything links fine with gcc in this situation
+# and everything links fine with clang when LLVM_ENABLE_ASSERTIONS is OFF. In order to
+# support the build with clang until this can be resolved, we set LLVM_ENABLE_ASSERTIONS
+# to OFF before including HandleLLVMOptions. This does not reproduce with clang on
+# Windows.
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND (NOT CMAKE_BUILD_TYPE STREQUAL "Debug"))
```

Comment:
@sstamenova  How should we resolve this?

Reply:
This is not trivial to fix (a.k.a. I haven't figured it out yet despite several hours spent on it). Unless you definitely want LLVM_ENABLE_ASSERTIONS to be possible to be set to ON for onnx-mlir, I think it is OK for the moment until it can be resolved later. I don't think LLVM_ENABLE_ASSERTIONS is required for onnx-mlir.

Reply:
@caoimhinuibrian: I can actually remove this workaround in the flags PR - cleaning up the flags that were being set explicitly in the cmake lets clang work correctly with LLVM_ENABLE_ASSERTIONS=ON. So once this change is committed, I will update the flags PR to remove the workaround.